src/components/LibraryManageDialog.vue

```diff
@@ -0,0 +1,603 @@
+};
+
+const isUninstallable = (
+  engineId: EngineId,
+  library: BrandedDownloadableLibrary
+) => {
+  const installedLibrary = installedLibraries.value[engineId].find(
+    (installedLibrary) => installedLibrary.uuid === library.uuid
+  );
+  if (!installedLibrary) {
+    return false;
+  }
+  return installedLibrary.uninstallable;
+};
+
+const fetchStatuses = ref<
+  Record<EngineId, "fetching" | "success" | "error" | undefined>
+>({});
+
+// 選択中の話者
+const selectedSpeakers = ref<Record<LibraryId, SpeakerId>>({});
+
+// 選択中のライブラリ
+const selectedLibrary = ref(LibraryId(""));
+const selectLibraryAndSpeaker = (
+  libraryId: LibraryId,
+  speakerId: SpeakerId
+) => {
+  selectedLibrary.value = libraryId;
+  selectedSpeakers.value[libraryId] = speakerId;
+};
+
+const libraryInfoToCharacterInfos = (
+  engineId: EngineId,
+  libraryInfo: DownloadableLibrary | InstalledLibrary
+): CharacterInfo[] => {
+  return libraryInfo.speakers.map((speaker) => {
+    return {
+      portraitPath: base64ImageToUri(speaker.speakerInfo.portrait),
+      metas: {
+        speakerUuid: SpeakerId(speaker.speaker.speakerUuid),
+        speakerName: speaker.speaker.name,
+        styles: speaker.speaker.styles.map((style) => {
+          const styleInfo = speaker.speakerInfo.styleInfos.find(
+            (info) => info.id == style.id
+          );
+          if (styleInfo === undefined) {
+            throw Error("styleInfo === undefined");
+          }
+          return {
+            styleName: style.name,
+            styleId: StyleId(style.id),
+            iconPath: base64ImageToUri(styleInfo.icon),
+            portraitPath: styleInfo.portrait
+              ? base64ImageToUri(styleInfo.portrait)
+              : undefined,
+            engineId,
+            voiceSamplePaths: styleInfo.voiceSamples,
+          };
+        }),
+        policy: speaker.speakerInfo.policy,
+      },
+    };
+  });
+};
+
+watch(
+  modelValueComputed,
+  async (newValue) => {
+    if (!newValue) {
+      return;
+    }
+    const engineIdsWithNotFetchingDownloadableLibraries = [];
+    for (const engineId of engineIdsWithDownloadableLibraries.value) {
+      if (
+        fetchStatuses.value[engineId] === "fetching" ||
+        fetchStatuses.value[engineId] === "success"
+      ) {
+        continue;
+      }
+      fetchStatuses.value[engineId] = "fetching";
+      engineIdsWithNotFetchingDownloadableLibraries.push(engineId);
+    }
+    for (const engineId of engineIdsWithNotFetchingDownloadableLibraries) {
+      const fetchResult = await store
+        .dispatch("INSTANTIATE_ENGINE_CONNECTOR", {
+          engineId,
+        })
+        .then((instance) =>
+          Promise.all([
+            instance.invoke("downloadableLibrariesDownloadableLibrariesGet")(
+              {}
+            ),
+            instance.invoke("installedLibrariesInstalledLibrariesGet")({}),
+          ])
+        )
+        .then((libraryEndpoints): [
```

Comment:
```suggestion
        .then([fetchedDownloadableLibraries, fetchedInstalledLibraries]): [
```
こんな感じにすれば`[0]`とかをなくせて読みやすい気がします。